src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -794,4 +795,52 @@ public static void drawItemLayered(ItemRenderer renderer, IBakedModel modelIn, I
         }
         net.minecraftforge.client.ForgeHooksClient.setRenderLayer(null);
     }
+
+    // Execute the cached parsing of client message as a command. See net.minecraft.command.Commands#handleCommand for reference
+    // Return true leaves the message alone, false means it is supposed to be caught in net.minecraft.client.gui.screen.ChatScreen
+    public static boolean sendMessage(ParseResults<ISuggestionProvider> currentParse)
+    {
+        if(currentParse == null || currentParse.getReader().canRead()) return true;//Return only if parser fails
+        try
+        {
+            currentParse.getContext().getDispatcher().execute(currentParse);
+        }
+        catch (CommandException execution)
+        {
+            Minecraft.getInstance().player.sendMessage(new StringTextComponent("").func_230529_a_(execution.getComponent()).func_240699_a_(TextFormatting.RED), Util.field_240973_b_);
+        }
+        catch (CommandSyntaxException syntax)
+        {
+            if(syntax.getType() == CommandSyntaxException.BUILT_IN_EXCEPTIONS.dispatcherUnknownCommand())
+            {
+                return true;
+            }
+            Minecraft.getInstance().player.sendMessage(new StringTextComponent("").func_230529_a_(TextComponentUtils.toTextComponent(syntax.getRawMessage())).func_240699_a_(TextFormatting.RED), Util.field_240973_b_);
+            if (syntax.getInput() != null && syntax.getCursor() >= 0)
+            {
+                int position = Math.min(syntax.getInput().length(), syntax.getCursor());
+                IFormattableTextComponent details = new StringTextComponent("").func_240699_a_(TextFormatting.GRAY);
+                details.getStyle().func_240715_a_(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, currentParse.getReader().getString()));
+                if (position > 10)
+                {
+                    details.func_240702_b_("...");
+                }
+                details.func_240702_b_(syntax.getInput().substring(Math.max(0, position - 10), position));
+                if (position < syntax.getInput().length())
+                {
+                    details.func_230529_a_(new StringTextComponent(syntax.getInput().substring(position)).func_240701_a_(TextFormatting.RED, TextFormatting.UNDERLINE));
+                }
+                details.func_230529_a_(new TranslationTextComponent("command.context.here").func_240701_a_(TextFormatting.RED, TextFormatting.ITALIC));
+                Minecraft.getInstance().player.sendMessage(new StringTextComponent("").func_230529_a_(details).func_240699_a_(TextFormatting.RED), Util.field_240973_b_);
+            }
+        }
+        catch (Exception generic)
+        {
+            StringTextComponent message = new StringTextComponent(generic.getMessage() == null ? generic.getClass().getName() : generic.getMessage());
```

Comment:
IMO any non-command exception thrown by a client command should be logged

Reply:
This is a paste from the vanilla Commands class error handling.
Behavior is intentionally identical with server commands.